### PR TITLE
introduce prettier

### PR DESCRIPTION
**Pull request by hiroppy**
- [ ] This is a **bugfix**
- [ ] This is a **feature**
- [x] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### Motivation / Use-Case

I would like to introduce `prettier` for development.
And I picked configure file and command from webpack repository.

### Additional Info

Currently, the webpack-dev-server code is far out of webpack's rule and there are many changes in this PR.
So should we decide on a custom rule? /cc @evilebottnawi

### CODE REVIEW

**Comment by montogeek** [comment]
Code should have same tab/space width, that is current style.

File: .prettierrc.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+	tabWidth: 2,
+	overrides: [
+		{
+			files: "*.json",
+			options: {
+				useTabs: false
+			}
+		}
+	]
+};
```
Comment (evilebottnawi) [inline .prettierrc.js]: Use this config:
```
{
  "singleQuote": true,
  "trailingComma": "es5",
  "arrowParens": "always"
}
```

Comment (hiroppy) [reply .prettierrc.js]: Added,PTAL

Comment (hiroppy) [reply .prettierrc.js]: @evenstensberg I made a forced push to add `html` to prettier.

file: https://github.com/webpack/webpack-dev-server/blob/0185f6397e0edd0285fc2154d7e8fe94e9103f51/.prettierrc.js

Comment (evenstensberg) [reply .prettierrc.js]: Looks like a regular config, got no opinion here. @evilebottnawi has worked with this more to have a better argument

Comment (hiroppy) [reply .prettierrc.js]: @evenstensberg  Oops, Sorry, I mistook the name of the destination 🙇

**Comment by hiroppy** [comment]
@evilebottnawi sorry but do you want spaces or tabs? I’m fine either way!

**Comment by evilebottnawi** [comment]
@hiroppy spaces

**Comment by hiroppy** [comment]
@evilebottnawi I changed spaces from tabs:)

File: client-src/live/live.html [hunk 2]
```diff
@@ -1 +1,17 @@
...
+      content="width=device-width, height=device-height, initial-scale=1.0, user-scalable=no, minimum-scale=1.0, maximum-scale=1.0"
+    />
+    <script
+      type="text/javascript"
+      charset="utf-8"
+      src="/__webpack_dev_server__/live.bundle.js"
+    ></script>
+  </head>
+  <body></body>
+</html>
```
Comment (evilebottnawi) [inline client-src/live/live.html]: Let's don't use prettier here

**Comment by hiroppy** [comment]
@evilebottnawi I removed `html` from prettier. PTAL 🙏

**Pull request merged**